File: Ex1/sanntidEx1.py

from threading import Thread

i = 0
	


def minus():
	global i
	for j in range(1000000):
		i-=1
		print "minus gives: " , i, "\n"
	


def plus():
	global i
	for j in range(1000000):
		i+=1
		print "plus gives: " , i, "\n"
	
	

def main():
    mThread = Thread(target = minus, args = (),)
    pThread = Thread(target = plus, args = (),)

    pThread.start()
    mThread.start()

    pThread.join()
    mThread.join()

    print "done gives: " , i, "\n"


main()

File: Ex1/sanntidEx1.c

/*
'global shared int i = 0

main:
    spawn thread_1
    spawn thread_2
    join all threads (wait for them to finish)
    print i

thread_1:
    do 1_000_000 times:
        i++
thread_2:
    do 1_000_000 times:
        i--
*/

#include <pthread.h>
#include <stdio.h>

int i = 0;

void minus(){
	for (int j = 0; j< 1000000;j++){
		i--;
		//printf("minus gives: %i\n", i);
	}
}

void plus(){
	for (int j = 0; j< 1000000;j++){
		i++;
		//printf("plus gives: %i\n", i);
	}
}



int main(){
    pthread_t mThread;
    pthread_t pThread;

    pthread_create(&pThread, NULL, plus, NULL);
    pthread_create(&mThread, NULL, minus, NULL);
    
    pthread_join(pThread, NULL);
    pthread_join(mThread, NULL);
    printf("done gives: %i\n", i);
    return 0;
}